shared-libs/transitions/src/transitions/registration.js

```diff
@@ -75,6 +72,181 @@ const booleanExpressionFails = (doc, expr) => {
   return result;
 };
 
+const getDOB = doc => {
+  if (!doc || !doc.fields) {
+    return '';
+  }
+  const reportedDate = moment(doc.reported_date).startOf('day');
+  const years = parseInt(getYearsSinceDOB(doc), 10);
+  if (!isNaN(years)) {
+    return reportedDate.subtract(years, 'years');
+  }
+  const months = parseInt(getMonthsSinceDOB(doc), 10);
+  if (!isNaN(months)) {
+    return reportedDate.subtract(months, 'months');
+  }
+  const weeks = parseInt(getWeeksSinceDOB(doc), 10);
+  if (!isNaN(weeks)) {
+    return reportedDate.subtract(weeks, 'weeks');
+  }
+  const days = parseInt(getDaysSinceDOB(doc), 10);
+  if (!isNaN(days)) {
+    return reportedDate.subtract(days, 'days');
+  }
+  // no given date of birth - return reportedDate as it's the best we can do
+  return reportedDate;
+};
+
+const getYearsSinceDOB = doc => {
+  const fields = ['years_since_dob', 'years_since_birth', 'age_in_years'];
+  return findFirstDefinedValue(doc, fields);
+};
+
+const getMonthsSinceDOB = doc => {
+  const fields = ['months_since_dob', 'months_since_birth', 'age_in_months'];
+  return findFirstDefinedValue(doc, fields);
+};
+
+const getWeeksSinceDOB = doc => {
+  const fields = [
+    'weeks_since_dob',
+    'dob',
+    'weeks_since_birth',
+    'age_in_weeks',
+  ];
+  return findFirstDefinedValue(doc, fields);
+};
+
+const getDaysSinceDOB = doc => {
+  const fields = ['days_since_dob', 'days_since_birth', 'age_in_days'];
+  return findFirstDefinedValue(doc, fields);
+};
+
+/*
+ * Given a doc get the LMP value as a number, including 0.  Supports three
+ * property names atm.
+ * */
+const getWeeksSinceLMP = doc => {
+  const props = ['weeks_since_lmp', 'last_menstrual_period', 'lmp'];
+  const prop = props.find(prop => !isNaN(Number(doc.fields && doc.fields[prop])));
+  return Number(prop && doc.fields[prop]);
+};
```

Comment:
Sorry to pick at this nit but this ends up parsing the number three times (if you include line 136). How do you feel about...

```js
const getWeeksSinceLMP = doc => {
  const props = ['weeks_since_lmp', 'last_menstrual_period', 'lmp'];
  for (const prop of props) {
    const value = Number(doc.fields && doc.fields[prop]);
    if (!isNaN(value)) {
      return value;
    }
  }
};
```

And then remove the `Number()` from line 136.

Reply:
Sure, this works for me!